pandapipes/component_models/water_tower_component.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2020 by Fraunhofer Institute for Energy Economics
+# and Energy System Technology (IEE), Kassel. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be found in the LICENSE file.
+
+import numpy as np
+from numpy import dtype
+
+from pandapipes.component_models.abstract_models import NodeElementComponent
+from pandapipes.constants import GRAVITATION_CONSTANT, P_CONVERSION
+from pandapipes.idx_branch import FROM_NODE, TO_NODE, LOAD_VEC_NODES
+from pandapipes.idx_node import PINIT, LOAD, NODE_TYPE, P, EXT_GRID_OCCURENCE, HEIGHT
+from pandapipes.internals_toolbox import _sum_by_group
+from pandapipes.pipeflow_setup import get_lookup
+
+try:
+    import pplog as logging
+except ImportError:
+    import logging
+
+logger = logging.getLogger(__name__)
+
+
+class WaterTower(NodeElementComponent):
+    """
+
+    """
+
+    @classmethod
+    def table_name(cls):
+        return "water_tower"
+
+    @classmethod
+    def sign(cls):
+        return 1.
+
+    @classmethod
+    def create_pit_node_entries(cls, net, node_pit, node_name):
+        """
+        Function which creates pit node entries.
+
+        :param net: The pandapipes network
+        :type net: pandapipesNet
+        :param node_pit:
+        :type node_pit:
+        :return: No Output.
+        """
+        water_towers = net[cls.table_name()]
+        density = net.fluid.get_density(water_towers.t_k.values)
+        junction_idx_lookups = get_lookup(net, "node", "index")[node_name]
+        junction = cls.get_connected_junction(net)
+        index_wt = junction_idx_lookups[junction]
+        press = density * (water_towers.height_m.values - node_pit[index_wt, HEIGHT]) * \
+                GRAVITATION_CONSTANT / P_CONVERSION * water_towers.in_service.values
```

Comment:
What exactly is the definition of height_m? My guess was the height of the water column in the tower. However, if the junction's height is subtracted, this could lead to negative height, couldn't it?
I'd find it confusing if height_m was in "meters above sea level" but I do not know the conventions.
As a suggestion, maybe it should be renamed to tower_height_m. A scaling factor could be introduced to derive water_column_height_m which then serves as input for the static pressure (rho*g*water_column_height_m)

Reply:
It actually should be the height above sea level. But I agree. It really seems confusing. It is a convention in SINCAL. But I would change it in our case taking only the height of the water_column.

Reply:
Thanks for the adaption! However, thinking about it twice, maybe this leads to a problem regarding communicating vessels. (see my latest comment.) This could be the reason why Sincal uses absolute height.